lib/pageflow/file_types.rb

```diff
@@ -16,10 +18,29 @@ def find_by_collection_name!(collection_name)
       end || raise(FileType::NotFoundError, "No file type found for collection name '#{collection_name}'.")
     end
 
+    def find_by_model!(model)
+      detect do |file_type|
+        file_type.model == model
+      end || raise(FileType::NotFoundError, "No file type found for '#{model.name}'.")
+    end
+
     def with_thumbnail_support
       select do |file_type|
         file_type.model.instance_methods.include?(:thumbnail_url)
       end
     end
+
+    private
+
+    def search_for_nested_file_types(higher_level_file_types)
+      nested_file_types = []
+      higher_level_file_types.each do |file_type|
+        if file_type.nested_file_types.any?
+          nested_file_types += file_type.nested_file_types
+          nested_file_types += search_for_nested_file_types(file_type.nested_file_types)
+        end
+      end
+      nested_file_types.flatten.uniq(&:model)
```

Comment:
The `any?` check is not needed. Let's use `map` instead of `each` to remove the need for the `nested_file_types` local. If we make sure that the items from `higher_level_file_types` are included in the result, we can get rid of the extra array concatenation and `uniq` call in the `each` method above.